src/yelp-application.c

```diff
@@ -246,9 +294,11 @@ yelp_application_cmdline (GApplication     *app,
             cur = g_get_current_dir ();
             base = g_file_new_for_path (cur);
             new = g_file_resolve_relative_path (base, (*arguments)[i]);
-            newuri = g_file_get_uri (new);
-            g_free ((*arguments)[i]);
-            (*arguments)[i] = newuri;
+            if (g_file_query_exists (new, NULL)) {
```

Comment:
Maybe add a comment about it being a hack for handling the `--gapplication-service` flag which is unlikely to exist as a file in the current working directory?